src/local-git/GitLocalInterface.ts

```diff
@@ -0,0 +1,67 @@
+import { Commit, Repository, NavigatorBackend } from "../NavigatorBackendType";
+const nodegit = require("nodegit");
+
+export class GitLocal implements NavigatorBackend {
+  getRepositoryInformation(
+    repoPath: string,
+  ): Promise<{ repo: Repository; remoteRepoInfoPromise: Promise<Repository> }> {
+    throw new Error("Method not implemented.");
+  }
+
+  //Actions
+  async createOrUpdateBranchesForCommitStack(
+    repoPath: string,
+    commitStack: Commit[],
+  ): Promise<Commit[]> {
+    try {
+      const repo = this.getRepositoryInformation(repoPath);
+      for (let i = 0; i < commitStack.length; i++) {
+        const commitMessage = commitStack[i].title.replace(" ", "-"); //first word of the commit message's line
+        const newBranchName = `${commitMessage}-${i}`; //branch name is commit message -
```

Comment:
Hmm this is going to be weird haha. Ideally we'll want to get the feature branch's name and append to that instead. If we can't do that easily, maybe we'll want to punt this to the user? I can ask the user for the base string and pass that to this function 😆

Reply:
I'll look into getting the fist commit's branchname and append to that? 😆 Is that always the feature's branch name?

Reply:
> Is that always the feature's branch name?

Unfortunately probably not LOL, it could be `master` or `origin/feat/whatever`. It may be too difficult to identify a sane branch name to use

Reply:
I see, so do you want to go with the second solution 

> If we can't do that easily, maybe we'll want to punt this to the user? I can ask the user for the base string and pass that to this function 😆

Reply:
Yeah let's do that, it'll be the simplest solution I hope